cpmpy/transformations/linearize.py

```diff
@@ -572,3 +603,34 @@ def only_positive_coefficients(lst_of_expr):
 
     return newlist
 
+def linearize_objective(expr, supported=frozenset(["sum","wsum"])):
+    """
+    Takes a linear expression and outputs:
+    
+    - Decision variable: Var
+    - Linear: sum([Var])                                   (CPMpy class 'Operator', name 'sum')
+              wsum([Const],[Var])                          (CPMpy class 'Operator', name 'wsum')
+
+    Only difference with flatten_objective is that no negated boolvars are allowed.
+    """
+    
+    flatexpr, flatcons = flatten_objective(expr, supported=supported)
+    
+    if isinstance(flatexpr, Operator) and flatexpr.name not in {"sum","wsum"}:
+        if flatexpr.name not in supported:
+            raise Exception(f"{flatexpr} is not linear or is not supported. Please report on github")
+        # other operators in expression such as "min", "max"
+        posexpr = copy.copy(flatexpr)
```

Comment:
also here please first check if there are NegBoolViews at all, and only copy in that case.

and your ifs on line 619 and 631 are so related but yet trickily different; better
'if isinsatnce(...) or flatexpr.name in {"sum", "wsum"}:
  ...only_pos_bv_sub()...
else:
   ...the manual code...